text/src/autogluon/text/automm/data/process_text.py

```diff
@@ -0,0 +1,262 @@
+
+        if self.insert_sep:
+            max_length = self.max_len - (len(text_tokens) + 1)
+        else:
+            max_length = self.max_len - 2
+        trimmed_lengths = self.get_trimmed_lengths(
+            [len(txt_token) for txt_token in text_tokens],
+            max_length,
+            do_merge=True,
+        )
+        seg = 0
+        token_ids = [self.cls_token_id]
+        segment_ids = [seg]
+        for txt_token, trim_length in zip(text_tokens, trimmed_lengths):
+
+            if self.stochastic_chunk:
+                start_ptr = np.random.randint(0, len(txt_token) - trim_length + 1)
+            else:
+                start_ptr = 0
+            token_ids.extend(txt_token[start_ptr:(start_ptr + trim_length)].tolist())
+            segment_ids.extend([seg] * trim_length)
+            if self.insert_sep:
+                token_ids.append(self.sep_token_id)
+                segment_ids.append(seg)
+            seg = (seg + 1) % self.text_segment_num
+
+        if token_ids[-1] != self.sep_token_id:
+            token_ids.append(self.sep_token_id)
+            segment_ids.append(seg)
+
+        return {
+            f"{self.prefix}_{TEXT_TOKEN_IDS}": np.array(token_ids, dtype=np.int32),
+            f"{self.prefix}_{TEXT_VALID_LENGTH}": len(token_ids),
+            f"{self.prefix}_{TEXT_SEGMENT_IDS}": np.array(segment_ids, dtype=np.int32)
+        }
+
+    def build_one_token_sequence_from_text(
+            self,
+            text: List[str],
+    ) -> dict:
+        # tokenize text
+        tokens = []
+        warnings.filterwarnings(
+            "ignore",
+            "Token indices sequence length is longer than.*result in indexing errors"
+        )
+        for col_text in text:
+            col_tokens = self.tokenizer.encode(
+                col_text,
+                add_special_tokens=False,
+                truncation=False,
+            )
+            tokens.append(np.array(col_tokens, dtype=np.int32))
+
+        # build token sequence
+        return self.build_one_token_sequence(tokens)
+
+    @staticmethod
+    def get_special_tokens(tokenizer):
+        cls_id, sep_id = tokenizer.cls_token_id, tokenizer.sep_token_id
+        if cls_id is None or sep_id is None:
+            cls_id, sep_id = tokenizer.bos_token_id, tokenizer.eos_token_id
+        if cls_id is None or sep_id is None:
+            raise ValueError(
+                f"tokenizer class: {tokenizer.__class__.__name__} has no valid cls and sep ids."
+            )
+        return cls_id, sep_id
+
+    @staticmethod
+    def get_pretrained_tokenizer(
+            tokenizer_name: str,
+            checkpoint_name: str,
+    ):
+        tokenizer_class = ALL_TOKENIZERS[tokenizer_name]
+        if torch.distributed.is_initialized():
+            if torch.distributed.get_rank() == 0:
+                tokenizer_class.from_pretrained(checkpoint_name)
+            torch.distributed.barrier()
+        return tokenizer_class.from_pretrained(checkpoint_name)
```

Comment:
`tokenizer_class.from_pretrained(checkpoint_name)` would be called twice if `.get_rank()` returns 0. Is this expected?

Reply:
@zhiqiangdon I think we may simply remove the code related to `torch.distributed.get_rank() == 0`. These are not useful.

Reply:
Removed.